Yoast/wordpress-seo
# Improved FAQ block performance

## Summary

This PR can be summarized in the following changelog entry:

* Improves FAQ block performance.

## Relevant technical choices:

* Implemented `shouldComponentUpdate` in `Question` component, and used a shallow compare to prevent all steps from re-rendering when a single step is touched.
* Replaced all anonymous function declarations in the render functions with class member functions (this improves performance overall and was also necessary to make the shallow compare work in `Question`).

## Test instructions

This PR can be tested by following these steps:

* Keep your console open to catch potential errors.
* Create a FAQ with multiple questions.
* Make sure you can still switch questions with the up and down buttons.
* Make sure you can still edit all fields.

## Quality assurance

* [] I have added unittests to verify the code works as intended

Fixes #11787
